java/rocksjni/env.cc

```diff
@@ -77,3 +79,33 @@ void Java_org_rocksdb_RocksMemEnv_disposeInternal(
     JNIEnv* env, jobject jobj, jlong jhandle) {
   delete reinterpret_cast<rocksdb::Env*>(jhandle);
 }
+
+/*
+ * Class:     org_rocksdb_RocksHdfsEnv
+ * Method:    createHdfsEnv
+ * Signature: ()J
+ */
+jlong Java_org_rocksdb_RocksHdfsEnv_createHdfsEnv(
+    JNIEnv* env, jclass jclazz, jstring fsname) {
+  const char *fsNameNative = env->GetStringUTFChars(fsname, NULL);
+  rocksdb::Env* pHdfsEnv = nullptr;
+  rocksdb::Env** ppHdfsEnv = &pHdfsEnv;
+  rocksdb::Status s = rocksdb::NewHdfsEnv(ppHdfsEnv,
+    std::string(fsNameNative));
+  jlong handle = 0;
+  if (s.ok()) {
+    handle = reinterpret_cast<jlong>(*ppHdfsEnv);
+  }
+  env->ReleaseStringUTFChars(fsname, fsNameNative);
+  return handle;
+}
+
+/*
+ * Class:     org_rocksdb_RocksMemEnv
+ * Method:    disposeInternal
+ * Signature: (J)V
+ */
+void Java_org_rocksdb_RocksHdfsEnv_disposeInternal(
+    JNIEnv* env, jobject jobj, jlong jhandle) {
+  delete reinterpret_cast<rocksdb::Env*>(jhandle);
```

Comment:
It could be helpful for debugging to place an assert here, e.g.:

```C++
auto* rocks_env = reinterpret_cast<rocksdb::Env*>(jhandle);
assert(rocks_env);
delete rocks_env;
```